eZ/Publish/Core/Limitation/ContentTypeLimitationType.php

```diff
@@ -19,14 +19,15 @@
 use eZ\Publish\Core\Base\Exceptions\InvalidArgumentType;
 use eZ\Publish\API\Repository\Values\User\Limitation\ContentTypeLimitation as APIContentTypeLimitation;
 use eZ\Publish\API\Repository\Values\User\Limitation as APILimitationValue;
-use eZ\Publish\SPI\Limitation\Type as SPILimitationTypeInterface;
+use eZ\Publish\SPI\Limitation\Target;
+use eZ\Publish\SPI\Limitation\TargetAwareType as SPITargetAwareLimitationType;
 use eZ\Publish\Core\FieldType\ValidationError;
 use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
 
 /**
  * ContentTypeLimitation is a Content limitation.
  */
-class ContentTypeLimitationType extends AbstractPersistenceLimitationType implements SPILimitationTypeInterface
+class ContentTypeLimitationType extends AbstractPersistenceLimitationType implements SPITargetAwareLimitationType
```

Comment:
I'm worried a bit about BC when changing type of a limitation. We did this with `LanguageLimitation` because we accepted BC breaks as Limitation was not useful. In this case it will result in some targets not being passed when they should.

Reply:
In some point of view, ContentTypeLimitation is also not useful, in context of UI when user wants create content.